src/assets/src/components/preferences.tsx

```diff
@@ -69,25 +78,93 @@ function PreferencesEditor(props: PreferencesEditorProps) {
             label="As a host, I want to be notified via SMS when someone joins my empty queue." />
     );
 
-    const validateAndSubmit = (e: React.SyntheticEvent) => {
+    const oneTimePasswordTimer = () => {
+        let timer = 30;
+        const interval = setInterval(() => {
+            setTimeToResendCode(timer--);
+            if (timer === -1) {
+                clearInterval(interval);
+            }
+        }, 1000);
+
+        return () => clearInterval(interval);
+    }
+
+    const getOneTimePassword = async (e: React.SyntheticEvent) => {
+        e.preventDefault(); // Prevent page reload
+        if (sendingCode || verifying) return;
+        setSendingCode(true);
+
+        // check if time buffer has passed, phone number has changed, 
+        // and phone number is valid
+        const timeRemaining = timeToResendCode ?
+            [new Error(`You must wait ${timeToResendCode} more seconds before requesting a new verification code.`)]
+            : [];
+        const phoneChangedError = !changedPhoneNumber ?
```

Comment:
instead of showing the error message, can we disable/hide the "Get Verification Code" button if user did not enter a phone number or if the phone number is the same as the one on file?